src/test/java/uk/gov/hmcts/reform/blobrouter/services/storage/LeaseMetaDataCheckerTest.java

```diff
@@ -0,0 +1,119 @@
+
+@ExtendWith(MockitoExtension.class)
+@SuppressWarnings("unchecked")
+class LeaseMetaDataCheckerTest {
+
+    @Mock private BlobClient blobClient;
+    @Mock private BlobProperties blobProperties;
+
+    private Map<String, String> blobMetaData;
+
+    private static final String leaseId = "lease-id";
+
+    private static final String LEASE_EXPIRATION_TIME = "leaseExpirationTime";
+
+    private LeaseMetaDataChecker leaseMetaDataChecker;
+
+    @BeforeEach
+    void setUp() {
+        leaseMetaDataChecker = new LeaseMetaDataChecker(15);
+        blobMetaData = new HashMap<>();
+    }
+
+    @Test
+    void should_return_true_when_no_expiry_in_metadata() {
+        //given
+        given(blobClient.getProperties()).willReturn(blobProperties);
+        given(blobProperties.getMetadata()).willReturn(blobMetaData);
+
+        //when
+        boolean isReady = leaseMetaDataChecker.isReadyToUse(blobClient, leaseId);
+
+        //then
+        assertThat(isReady).isTrue();
+        var conditionCapturer = ArgumentCaptor.forClass(BlobRequestConditions.class);
+        verify(blobClient).setMetadataWithResponse(any(),conditionCapturer.capture(), any(), any());
+        assertThat(conditionCapturer.getValue().getLeaseId()).isEqualTo(leaseId);
+
+    }
+
+    @Test
+    void should_return_false_when_expiry_in_metadata_valid() {
+        //given
+        given(blobClient.getProperties()).willReturn(blobProperties);
+        blobMetaData.put(LEASE_EXPIRATION_TIME, LocalDateTime.now(EUROPE_LONDON_ZONE_ID).plusSeconds(40).toString());
+        given(blobProperties.getMetadata()).willReturn(blobMetaData);
+
+        //when
+        boolean isReady = leaseMetaDataChecker.isReadyToUse(blobClient, leaseId);
+
+        //then
+        assertThat(isReady).isFalse();
+        verify(blobClient,never()).setMetadataWithResponse(any(), any(), any(), any());
+
+    }
+
+    @Test
+    void should_return_true_when_metadata_lease_expiration_expired() {
+        //given
+        given(blobClient.getProperties()).willReturn(blobProperties);
+        blobMetaData.put(LEASE_EXPIRATION_TIME, LocalDateTime.now(EUROPE_LONDON_ZONE_ID).toString());
+
+        given(blobProperties.getMetadata()).willReturn(blobMetaData);
+
+        //when
+        boolean isReady = leaseMetaDataChecker.isReadyToUse(blobClient, leaseId);
+
+        //then
+        assertThat(isReady).isTrue();
+        LocalDateTime leaseExpiresAt = LocalDateTime.parse(blobMetaData.get(LEASE_EXPIRATION_TIME));
+        assertThat(leaseExpiresAt.isAfter(LocalDateTime.now(EUROPE_LONDON_ZONE_ID))).isTrue();
+        var conditionCapturer = ArgumentCaptor.forClass(BlobRequestConditions.class);
+        verify(blobClient).setMetadataWithResponse(any(),conditionCapturer.capture(), any(), any());
+        assertThat(conditionCapturer.getValue().getLeaseId()).isEqualTo(leaseId);
+
+    }
+
+    @Test
+    void should_clear_blob_metadata_when_clear_successful() {
```

Comment:
Makes sense to ensure that only lease expiration metadata is cleared and not any other if it has been also set.

Reply:
but that time it needs to reread the while metadata

Reply:
The test can add some other metadata other than lease expiration time and ensure that this extra metadata is still there after cleaning up.

Reply:
it clears all metada not just lease expiration time

Reply:
I have approved but if some other metadata will be used in the future - this clean up operation will delete it as well and this will be wrong